src/coreclr/jit/gentree.h

```diff
@@ -2244,13 +2244,12 @@ struct GenTree
 #if defined(TARGET_XARCH) && defined(FEATURE_HW_INTRINSICS)
     bool IsEmbMaskOp()
     {
-        bool result = (gtFlags & GTF_HW_EM_OP) != 0;
-        assert(!result || (gtOper == GT_HWINTRINSIC));
-        return result;
+        return OperIsHWIntrinsic() && ((gtFlags & GTF_HW_EM_OP) != 0);
```

Comment:
None of the callsites were checking `OperIsHWIntrinsic()` first, so needed to make the check explicit.

Asserting in `MakeEmbMaskOp` was then appropriate since it did check first.